src/sage/geometry/polyhedral_complex.py

```diff
@@ -2356,7 +2368,29 @@ def subdivide(self, make_simplicial=False,
             sage: subdiv_halfspace.is_simplicial_fan()
             True
         """
+        if isinstance(weights, collections.abc.Mapping):
+            weights = {vector(v, immutable=True): weight for v, weight in weights.items()}
+            weights = weights.__getitem__
         if self.is_compact():
+            if weights:
+                def regular_subdivision(polytope):
+                    vertices = set(tuple(v.vector()) for v in polytope.vertex_generator())
+                    if new_vertices:
```

Comment:
Maybe the new vertices should not have an assignated weight, because first we only do regular subdiv and only then we do further subdivisions.

Reply:
new vertices could have some "infinity" (very large) so that they do not play any role at this stage